Source Packages/java/user/User.java

```diff
@@ -1373,6 +1345,163 @@ public boolean isOldDrupalUser() throws SQLException        {
         }
     }
     
-}
+    /**Get the username (part of email address before @) of the owner of this comment
+     * @return
+     */
+    public String getUname()
+        {
+        try
+            {
+
+            return openID.substring(0, openID.indexOf('@'));
+            } catch (Exception e)
+            {
+            return Uname;
+            }
+        }
+
+    /**Get the first name of the owner of this comment
+     * @return
+     */
+    public String getFname()
+        {
+        return fname;
+        }
+
+    /**Get the first name of the owner of this comment
+     * @return
+     */
+    public String getLname()
+        {
+        return lname;
+        }
+    /**Get the email of the owner of this comment
+     * @return
+     */
+    public String getEmail()
+        {
+        return email;
+        }
     
+    /**
+     * Change the username for this user in the db.
+     */
+    public int changeUsername(String username){
```

Comment:
Yeah I had it as that initially, but it is not a typical set which would be like

```java
 public int setUsername(String username){
      this.Uname = username
 }
```

so it was confusing me.  I named it with `change` because you are never using this function to set a username.  You already have a username in the User initiated by the class.  What you are trying to do is change it.  I would be OK with renaming it, but it is not a typical `setUname()`.